apps/docs/docs/modules/agent/openai.mdx

```diff
@@ -0,0 +1,215 @@
+# OpenAI Agent
+
+OpenAI API that supports function calling, it’s never been easier to build your own agent!
+
+In this notebook tutorial, we showcase how to write your own OpenAI agent
+
+## Setup
+
+First, you need to install the `llamaindex` package. You can do this by running the following command in your terminal:
+
+```bash
+pnpm i llamaindex
+```
+
+Then we can define a function to sum two numbers and another function to divide two numbers.
+
+```ts
+function sumNumbers({ a, b }: { a: number; b: number }): number {
+  return a + b;
+}
+
+// Define a function to divide two numbers
+function divideNumbers({ a, b }: { a: number; b: number }): number {
+  return a / b;
+}
+```
+
+## Create a function tool
+
+Now we can create a function tool from the sum function and another function tool from the divide function.
+
+For the parameters of the sum function, we can define a JSON schema or a Zod schema.
+
+### JSON Schema
+
+```ts
+const sumJSON = {
+  type: "object",
+  properties: {
+    a: {
+      type: "number",
+      description: "The argument a to sum",
+    },
+    b: {
+      type: "number",
+      description: "The argument b to sum",
+    },
+  },
+  required: ["a", "b"],
+};
+
+const divideJSON = {
+  type: "object",
+  properties: {
+    a: {
+      type: "number",
+      description: "The argument a to divide",
+    },
+    b: {
+      type: "number",
+      description: "The argument b to divide",
+    },
+  },
+  required: ["a", "b"],
+};
+
+const sumFunctionTool = new FunctionTool(sumNumbers, {
+  name: "sumNumbers",
+  description: "Use this function to sum numbers together",
+  parameters: sumJSON,
+});
+
+const divideFunctionTool = new FunctionTool(divideNumbers, {
+  name: "divideNumbers",
+  description: "Use this function to divide numbers together",
+  parameters: divideJSON,
+});
+```
+
+### Zod Schema
+
+```ts
+import { z } from "zod";
+
+const sumArgsSchema = z
+  .object({
+    a: z.number().describe("The argument a to sum"),
+    b: z.number().describe("The argument b to sum"),
+  })
+  .describe("the arguments");
+
+const dividArgsSchema = z
+  .object({
+    a: z.number().describe("The argument a to divide"),
+    b: z.number().describe("The argument b to divide"),
+  })
+  .describe("the arguments");
+
+const sumFunctionTool = new FunctionTool(sumNumbers, {
```

Comment:
So for example, sumNumbersZod could be a zod function schema https://zod.dev/?id=functions with a `.implement`, and then FunctionTool should only need to take in the single argument.